connectors/byoc.py

```diff
@@ -320,17 +341,35 @@ def has_advanced_rules(self):
 
 class PipelineSettings:
     def __init__(self, pipeline):
-        self.name = pipeline.get("name", "ent-search-generic-ingestion")
-        self.extract_binary_content = pipeline.get("extract_binary_content", True)
-        self.reduce_whitespace = pipeline.get("reduce_whitespace", True)
-        self.run_ml_inference = pipeline.get("run_ml_inference", True)
+        self._pipeline = pipeline
+        if self._pipeline is None:
+            self._pipeline = {}
+
+    @property
+    def name(self):
+        return self._pipeline.get("name", "ent-search-generic-ingestion")
+
+    @property
+    def extract_binary_content(self):
+        return self._pipeline.get("extract_binary_content", True)
+
+    @property
+    def reduce_whitespace(self):
+        return self._pipeline.get("reduce_whitespace", True)
+
+    @property
+    def run_ml_inference(self):
+        return self._pipeline.get("run_ml_inference", True)
 
     def __repr__(self):
         return (
             f"Pipeline {self.name} <binary: {self.extract_binary_content}, "
             f"whitespace {self.reduce_whitespace}, ml inference {self.run_ml_inference}>"
         )
 
+    def to_dict(self):
+        return dict(self._pipeline)
+
```

Comment:
I think it can be replaced with `__dict__` function and calling `dict` on instance of PipelineSettings should call it. cc @tarekziade is it correct?

Reply:
I don't think `__dict__` is for this purpose ([ref](https://docs.python.org/3/library/stdtypes.html#object.__dict__)):
>A dictionary or other mapping object used to store an object’s (writable) attributes.

Reply:
Sorry my bad. What I've meant is that if it's used as hash, then the class might actually expose dictionary functions so that it could be done outside of the object, e.g:

pipeline_dict = dict(Pipeline) or something else.

In short term this code works, but I believe we can improve on it later - in later PRs.